sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ResolveTableSpec.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.optimizer.ConstantFolding
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreePattern.COMMAND
+import org.apache.spark.sql.connector.catalog.CatalogManager
+
+/**
+ * This rule is responsible for analyzing expressions passed in as values for OPTIONS lists for
+ * commands such as CREATE TABLE. These expressions may be constant but non-literal, in which case
+ * we perform constant folding here.
+ */
+case class ResolveTableSpec(catalogManager: CatalogManager) extends Rule[LogicalPlan] {
```

Comment:
I wonder if we can avoid creating a new rule for this.
Probably all we need is to make `mapExpressions` supports the type `Map[String, Expression]` https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala#L217
With that, analyzer will analyze the expression and the optimizer will do the constant folding.

Reply:
Sounds good, done!